fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(data => this.setState({ todos: data }));
+  }
+
+  addTodo(newTodo) {
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ task: newTodo.task })
+    };
+
+    fetch('/api/todos', requestOptions)
+      .then(response => response.json())
+      .then(data => this.setState({ todos: this.state.todos.concat(data) })
+      );
+  }
+
+  toggleCompleted(todoId) {
+
+    const isCompletedToggleState = !this.state.todos[this.state.todos.findIndex(element => element.todoId === todoId)].isCompleted;
```

Comment:
If you saved the found `index` in a variable here you could use that same `index` in the `.then` and you wouldn't need to loop through the data twice to get the index.